v3-sdk/modifying-position/src/example/Example.tsx

```diff
@@ -0,0 +1,386 @@
+    tokenA: CurrentConfig.tokens.token0,
+    tokenB: CurrentConfig.tokens.token1,
+    fee: CurrentConfig.tokens.poolFee,
+  })
+
+  const poolContract = new ethers.Contract(
+    currentPoolAddress,
+    IUniswapV3PoolABI.abi,
+    provider
+  )
+
+  const [token0, token1, fee, tickSpacing, liquidity, slot0] =
+    await Promise.all([
+      poolContract.token0(),
+      poolContract.token1(),
+      poolContract.fee(),
+      poolContract.tickSpacing(),
+      poolContract.liquidity(),
+      poolContract.slot0(),
+    ])
+
+  return {
+    token0,
+    token1,
+    fee,
+    tickSpacing,
+    liquidity,
+    sqrtPriceX96: slot0[0],
+    tick: slot0[1],
+  }
+}
+
+const getPosition = async (
+  percentageToIncrease?: number
+): Promise<Position> => {
+  // get pool info
+  const poolInfo = await getPoolInfo()
+
+  // construct pool instance
+  const USDC_DAI_POOL = new Pool(
+    CurrentConfig.tokens.token0,
+    CurrentConfig.tokens.token1,
+    poolInfo.fee,
+    poolInfo.sqrtPriceX96.toString(),
+    poolInfo.liquidity.toString(),
+    poolInfo.tick
+  )
+
+  let amount0 = fromReadableAmount(
+    CurrentConfig.tokens.token0Amount,
+    CurrentConfig.tokens.token0.decimals
+  )
+  let amount1 = fromReadableAmount(
+    CurrentConfig.tokens.token1Amount,
+    CurrentConfig.tokens.token1.decimals
+  )
+
+  if (percentageToIncrease) {
+    amount0 = (amount0 * percentageToIncrease) / 100
+    amount1 = (amount1 * percentageToIncrease) / 100
+  }
+
+  // create position using the maximum liquidity from input amounts
+  return Position.fromAmounts({
+    pool: USDC_DAI_POOL,
+    tickLower:
+      nearestUsableTick(poolInfo.tick, poolInfo.tickSpacing) -
+      poolInfo.tickSpacing * 2,
+    tickUpper:
+      nearestUsableTick(poolInfo.tick, poolInfo.tickSpacing) +
+      poolInfo.tickSpacing * 2,
+    amount0,
+    amount1,
+    useFullPrecision: true,
+  })
+}
+
+async function mintPosition(): Promise<TransactionState> {
+  const address = getWalletAddress()
+  const provider = getProvider()
+  if (!address || !provider) {
+    return TransactionState.Failed
+  }
+  // Give approval to the contract to transfer tokens
+  const tokenInApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.token0.address,
+    address,
+    NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS
+  )
+  const tokenOutApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.token1.address,
+    address,
+    NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS
+  )
+
+  if (
+    tokenInApproval !== TransactionState.Sent ||
+    tokenOutApproval !== TransactionState.Sent
+  ) {
+    return TransactionState.Failed
+  }
+
+  // get calldata for minting a position
+  const { calldata, value } = NonfungiblePositionManager.addCallParameters(
+    await getPosition(),
+    {
+      recipient: address,
+      deadline: Math.floor(Date.now() / 1000) + 60 * 20,
+      slippageTolerance: new Percent(50, 10_000),
+    }
+  )
```

Comment:
nit: breaking this down here doesn't show the intermediary type / fetching well. Thoughts on giving this a name and using below instead? Can also add a type to show what the result is etc